sklearnex/preview/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,236 @@
+)
+class IncrementalPCA(sklearn_IncrementalPCA):
+
+    def __init__(self, n_components=None, *, whiten=False, copy=True, batch_size=None):
+        super().__init__(
+            n_components=n_components, whiten=whiten, copy=copy, batch_size=batch_size
+        )
+        self._need_to_finalize = False
+
+    _onedal_incremental_pca = staticmethod(onedal_IncrementalPCA)
+
+    def _onedal_transform(self, X, queue=None):
+        assert hasattr(self, "_onedal_estimator")
+        if self._need_to_finalize:
+            self._onedal_finalize_fit()
+        X = check_array(X, dtype=[np.float64, np.float32])
+        return self._onedal_estimator.predict(X, queue)
+
+    def _onedal_fit_transform(self, X, queue=None):
+        self._onedal_fit(X, queue)
+        return self._onedal_transform(X, queue)
+
+    def _onedal_partial_fit(self, X, check_input=True, queue=None):
+        first_pass = not hasattr(self, "components_")
+
+        if check_input:
+            if sklearn_check_version("1.0"):
+                X = self._validate_data(
+                    X, dtype=[np.float64, np.float32], reset=first_pass
+                )
+            else:
+                X = check_array(
+                    X,
+                    dtype=[np.float64, np.float32],
+                    copy=self.copy,
+                )
+
+        if not hasattr(self, "n_samples_seen_"):
+            self.n_samples_seen_ = 0
+
+        onedal_params = {"n_components": self.n_components, "whiten": self.whiten}
+
+        if not hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator = self._onedal_incremental_pca(**onedal_params)
+        self._onedal_estimator.partial_fit(X, queue)
+        self._need_to_finalize = True
+
+    def _onedal_finalize_fit(self):
+        assert hasattr(self, "_onedal_estimator")
+        self._onedal_estimator.finalize_fit()
+        self._need_to_finalize = False
+
+    def _onedal_fit(self, X, queue=None):
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(X, dtype=[np.float64, np.float32], copy=self.copy)
+        else:
+            X = check_array(
+                X,
+                dtype=[np.float64, np.float32],
+                copy=self.copy,
+            )
+
+        n_samples, n_features = X.shape
+
+        if self.batch_size is None:
+            self.batch_size_ = 5 * n_features
+        else:
+            self.batch_size_ = self.batch_size
+
+        self.n_samples_seen_ = 0
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator._reset()
+
+        for batch in gen_batches(X.shape[0], self.batch_size_):
+            X_batch = X[batch]
+            self._onedal_partial_fit(X_batch, queue=queue)
+
+        if X.shape[0] == 1:
+            warnings.warn(
+                "Only one sample available. You may want to reshape your data array"
+            )
```

Comment:
Is this oneDAL specific issue, if so should these be caught in the onedal_supported right?